core/PRP/Physics/plGenericPhysical.cpp

```diff
@@ -733,3 +735,39 @@ void plGenericPhysical::setTMDBuffer(size_t tmdSize, const unsigned char* tmdBuf
     fTMDBuffer = new unsigned char[fTMDSize];
     memcpy(fTMDBuffer, tmdBuffer, fTMDSize);
 }
+
+void plGenericPhysical::calcSphereBounds(size_t numPoints, const hsVector3* points) {
+    hsBounds3 bounds;
+    for (size_t i = 0; i < numPoints; ++i)
+        bounds += points[i];
+
+    hsVector3 distance = (bounds.getMaxs() - bounds.getMins()) * .5f;
+    fRadius = std::max(distance.X, std::max(distance.Y, distance.Z));
+    fOffset = bounds.updateCenter();
+}
+
+void plGenericPhysical::calcBoxBounds(size_t numPoints, const hsVector3* points) {
+    fIndices.clear();
+    fVerts.clear();
+
+    hsBounds3Ext bounds;
+    for (size_t i = 0; i < numPoints; ++i)
+        bounds += points[i];
+    bounds.unalign();
+
+    // PhysX and ODE can do true boxes
+    fDimensions = (bounds.getMaxs() - bounds.getMins()) * .5f;
+    fOffset = bounds.updateCenter();
+
+    // Havok wants actual geometry
+    fVerts.resize(8);
+    fVerts[0] = fVerts[1] = fVerts[2] = fVerts[4] = bounds.getCorner();
+    fVerts[1] += bounds.getAxis(0);
+    fVerts[2] += bounds.getAxis(1);
+    fVerts[4] += bounds.getAxis(2);
```

Comment:
It's been awhile since I wrote this code, so that's a good question. At one point, I hacked in code to generate faces for these verts and viewed the trimesh proxy in CWE, and the resulting geometry matched the visuals. Of course, that could have just been a fluke. IIRC I ported this from some code in CWE that renders bounding boxes.